src/App.js

```diff
@@ -1,24 +1,117 @@
 import './App.css';
+import React, { useState, useEffect } from "react";
+import {MdClear, MdAddCircle, MdCheckCircle} from 'react-icons/md'
+import{v4 as UuidV4} from "uuid";
 
-function App() {
+const App = () => {
+  /* 로컬 스토리지 저장 */
+  const initState = JSON.parse(localStorage.getItem("todos")) || [];
+  const [input, setInput] = useState("");
+  const [todos, setTodos] = useState(initState);
+  const [dones, setDones] = useState([]);
+
+  useEffect(()=>{
+    localStorage.setItem("todos", JSON.stringify(todos));
+  }, [todos]);
+
+/* todo length를 통해 아이템 카운트*/
+  return <div className='Background'>
+  <div className='do-list'>아직 완료 안된 일 ({todos.length})</div>
+  <div className='box-list'>
+  <UnDoneList todos={todos} setTodos = {setTodos}/>
+  </div>
+  <div className='do-list'>완료된 일 ({dones.length}) </div>
+  <div className='box-list'>
+  <DoneList  dones={dones} setDones = {setDones}/>
+  </div>
+  <Form 
+  input = {input}
+  setInput = {setInput}
+  todos = {todos}
+  setTodos = {setTodos}
+
+  />
+  </div>
+}
+
+/* 할일 입력 폼 구성*/
+const Form = ({input, setInput, todos, setTodos}) => {
+  const onInputChange = (event) =>{
+    setInput(event.target.value);
+  }
+  const onFormSubmit = (event) =>{
+    event.preventDefault();
+    setTodos([...todos, {id:UuidV4(), title:input, completed:false}]);
+    setInput("");
+  }
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+    <form onSubmit={onFormSubmit}>
+      <input type = 'text' placeholder='내용 입력하기...' className='task-input' value={input} required onChange={onInputChange}  />
+      <button className='button-add' type='submit'>{<MdAddCircle />}</button>
+    </form>
+  )
 }
+/* 완료 안된 일의 리스트 구성 */
+const UnDoneList = ({todos, setTodos, dones, setDones}) => {
+  const handleDelete = ({id})=>{
+    setTodos(todos.filter((todo)=> todo.id !==id));
+  }
+  const handleComplete = ({id})=>{
+    setTodos(todos.filter((todo)=> todo.id !==id));
+    setDones(dones.filter((done)=> done.id ===id));
```

Comment:
제가 보기에 이 부분때문에 완료가 안된 todo item을 완료했을때 완료 목록으로 넘어가지 않은 것 같습니다!
```suggestion
    const targetTodoItem = todos.filter((todo) => todo.id === id));
    setTodos(todos.filter((todo)=> todo.id !==id));
    setDones([...dones, targetTodoItem]);
```
현재 정수님의 코드는 완료될 todo item(개별 할일을 말합니다)을 `todos`에서 삭제하는 것 까지는 정상적으로 동작하나, `dones`에서 인자로 받은 `id`를 갖는 todo item 만을 필터링 합니다. 당연히 현재 인자로 받은 `id`를 갖는 todo item은 `dones`에 존재하지 않기 때문에 `dones.filter((done)=> done.id ===id)`의 결과는 `null`이 됩니다. 따라서 todo item이 토글되지 않았던 것이구요.

Reply:
한 마디 얹자면, setDones에는  **(원래 dones에 있던 item)들 + (이번에 done이 된 item)** 모두 넘겨 주셔야 정상적으로 구현이 가능할 것 같습니다!

Reply:
completed 값을 true로 바꿔주는 과정도 필요할 것 같아요😉